Runner/utils/functestlib.sh

```diff
@@ -91,3 +91,91 @@ functestlibdoc()
   done
   echo "Note, these functions will probably not work with >=32 CPUs"
 }
+
+# Function is to check for network connectivity status
+check_network_status() {
+    echo "[INFO] Checking network connectivity..."
+ 
+    # Get first active IPv4 address (excluding loopback)
+    ip_addr=$(ip -4 addr show scope global up | grep -oP '(?<=inet\s)\d+(\.\d+){3}' | head -n 1)
+ 
+    if [ -n "$ip_addr" ]; then
+        echo "[PASS] Network is active. IP address: $ip_addr"
+ 
+        if ping -c 1 -W 2 8.8.8.8 >/dev/null 2>&1; then
+            echo "[PASS] Internet is reachable."
+            return 1
+        else
+            echo "[WARN] Network active but no internet access."
+            return 0
+        fi
+    else
+        echo "[FAIL] No active network interface found."
+        return 0
+    fi
+}
+
+# If the tar file already exists,then function exit. Otherwise function to check the network connectivity and it will download tar from internet.
+extract_tar_from_url() {
+    local url="$1"
+    local filename
+    local extracted_files
+	
+	# Extract the filename from the URL
+    filename=$(basename "$url")
+	if check_tar_file "$filename"; then
+	   echo "[PASS] file already exists, Hence skipping downloading"
+	   return 1
```

Comment:
The return codes are really confusing. Sometimes `1` means OK and sometimes it means failure. I propose to return `0` always when there is no problem and use non-zero codes when there are issues.